crates/connectors/ndc-postgres/src/schema.rs

```diff
@@ -393,12 +406,83 @@ fn insert_to_procedure(
         },
     );
 
+    make_procedure_type(
+        name.to_string(),
+        Some(insert.description.to_string()),
+        arguments,
+        models::Type::Named {
+            name: insert.collection_name.to_string(),
+        },
+        object_types,
+        scalar_types,
+    )
+}
+
+/// Build a `ProcedureInfo` type from the given parameters.
+///
+/// Because procedures return an `affected_rows` count alongside the result type that it's
+/// `returning`, we have to generate a separate object type for its result. As part of that, we may
+/// also have to include the `int4` scalar type (if it isn't included for another reason elsewhere
+/// in the schema). So, this function creates that object type, optionally adds that scalar type,
+/// and then returns a `ProcedureInfo` that points to the correct object type.
+fn make_procedure_type(
+    name: String,
+    description: Option<String>,
+    arguments: BTreeMap<String, models::ArgumentInfo>,
+    result_type: models::Type,
+
+    object_types: &mut BTreeMap<String, models::ObjectType>,
+    scalar_types: &mut BTreeMap<String, models::ScalarType>,
+) -> models::ProcedureInfo {
+    let mut fields = BTreeMap::new();
+    let object_type_name = format!("{name}_response");
+
+    // If int4 doesn't exist anywhere else in the schema, we need to add it here. However, a user
+    // can't filter or aggregate based on the affected rows of a procedure, so we don't need to add
+    // any aggregate functions or comparison operators. However, if int4 exists elsewhere in the
+    // schema and has already been added, it will also already contain these functions and
+    // operators.
+    scalar_types
+        .entry("int4".to_string())
+        .or_insert(models::ScalarType {
+            aggregate_functions: BTreeMap::new(),
+            comparison_operators: BTreeMap::new(),
+        });
```

Comment:
Is this also required when generating the configuration?

If it is, then it will need to be moved to `occurring_scalar_types`.

Reply:
@soupi

Reply:
Thinking about it, even if it's not, that still feels like a better place for it. I plan on making that function private and putting the scalar types into the configuration at some point soon, so this function can just read them directly.

Reply:
It's not really part of the configuration. Even if the user omits `int4` from the configuration we still need to expose this type for the schema or else procedures will break, so I don't see a way out of defining it here.

Reply:
I am convinced.

Any reason not to add it in up-front rather than mutating later?

Reply:
Only that, if no mutations are generated, it's a type without a purpose, which we seem to be avoiding everywhere else? idk, I'm not sure what the danger is in having all scalar types declared upfront and not having the `ocurring_scalar_types` check, but doing it this way means we're still respecting that approach. Probably a wider question

Reply:
I think it's better to keep the cause and effect together. We generate the type because the procedure needs it, and we don't if we don't. It's better to place them together instead of in two separate places.